src/FeaturedComponents.jsx

```diff
@@ -1,78 +1,220 @@
-let components = [];
+const GRAPHQL_ENDPOINT = props.GRAPHQL_ENDPOINT || "https://near-queryapi.api.pagoda.co";
+const LIMIT = 8;
 const componentsUrl = "/${REPL_ACCOUNT}/widget/ComponentsPage";
 
+const sortRaw = Storage.get("queryapi:component-feed-sort");
+const intialSortOption = sortRaw === undefined ? 'featured' : sortRaw;
+
+const [components, setComponents] = useState([]);
+const [isLoading, setIsLoading] = useState(false);
+const [sortOption, setSortOption] = useState('');
+
 const featuredComponentListRes = Social.get("${REPL_FEATURED_COMP_MANAGER}/listManager/FeaturedComponents", "final");
 const featuredComponentPaths = featuredComponentListRes ? JSON.parse(featuredComponentListRes) : [];
-
 const componentData = featuredComponentPaths.length > 0 ? Social.getr(featuredComponentPaths) : null;
 
-if (componentData) {
-  featuredComponentPaths.forEach((src) => {
-    const path = src.split("/");
-    const result = {
-      metadata: componentData[path[0]][path[1]][path[2]].metadata,
-      src,
-    };
+const followGraph = context.accountId ? Social.keys(`${context.accountId}/graph/follow/*`, "final") : null;
+const accountsFollowing =
+  props.accountsFollowing ?? (followGraph ? Object.keys(followGraph[context.accountId].graph.follow || {}) : null);
 
-    components.push(result);
+const optionsMap = {
+    featured: "Featured",
+    followed: "From Followed",
+};
+
+function fetchGraphQL(operationsDoc, operationName, variables) {
+  return asyncFetch(`${GRAPHQL_ENDPOINT}/v1/graphql`, {
+    method: "POST",
+    headers: { "x-hasura-role": "kevin0_near" },
+    body: JSON.stringify({
+      query: operationsDoc,
+      variables: variables,
+      operationName: operationName,
+    }),
   });
 }
 
+const createQuery = () => {
+  console.log(accountsFollowing); //["roshaan.near", "calebjacob.near", "potlock.near"]
+  return `
+  query ComponentFollowingQuery($accountsFollowing: [String], $limit: Int) {
+    kevin0_near_component_01_info(
```

Comment:
Will need to replace `kevin0_near_component_01_info` once the indexer is updated.